### PR TITLE
Add baremetal ovn job using local gateway mode

### PR DESCRIPTION
**Pull request by jluhrsen**
Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

**Comment by jluhrsen** [comment]
this PR is meaningless until this [one in dev-scripts](https://github.com/openshift-metal3/dev-scripts/pull/1238) is merged

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.8.yaml
Comment (vrutkovs) [inline ci-operator/config/openshift/release/openshift-release-master__nightly-4.8.yaml:163]: Why the weird cron? Perhaps `interval` would make more sense?

Comment (jluhrsen) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.8.yaml:163]: oh yeah, that's my bad. I wanted it 48h. it's updated now. thanks for catching that.

**Comment by jluhrsen** [comment]
metal-ipi jobs do not work with rehearsals, so we have to merge this without seeing that. it's a new job though and I can monitor it in case something funky is broken.

**Comment by jluhrsen** [comment]
@vrutkovs , can we get this in please?

**Review by vrutkovs — APPROVED** [review]
/approve

**Comment by vrutkovs** [comment]
Seems you need someone else to approve that.
/cc wking smarterclayton

>metal-ipi jobs do not work with rehearsals

Wait, why?

**Comment by jluhrsen** [comment]
> Seems you need someone else to approve that.
> /cc wking smarterclayton
> 
> > metal-ipi jobs do not work with rehearsals
> 
> Wait, why?

because the config is too big and we hit some OS limit. I can't find one PR that I was more verbose about it,
but [here is one](https://github.com/openshift/release/pull/18473#issuecomment-837918376) where I was able to remove some steps to reduce the config and let the rehearse job run
the steps that I was changing to show the PR was going to work. It points to this [slack discussion](https://coreos.slack.com/archives/CBN38N3MW/p1620679486306000) as well.

**Comment by vrutkovs** [comment]
If it won't pass rehearsals now, it probably would be permafailing in periodics

/retest

**Comment by jluhrsen** [comment]
> If it won't pass rehearsals now, it probably would be permafailing in periodics
> 
> /retest

no, the periodics are fine, like [this](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-nightly-4.8-e2e-metal-ipi-ovn-ipv6/1394665725724987392), but here is what a [rehearsal](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/18473/rehearse-18473-pull-ci-openshift-cluster-network-operator-release-4.9-e2e-metal-ipi-ovn-ipv6-ipsec/1392002968739385344) looks like.

**Comment by stbenjam** [comment]
> If it won't pass rehearsals now, it probably would be permafailing in periodics
> 
> /retest

It's a problem with rehearsals only, the entire job contents including all step metadata and scripts are stored in an environment variable and we hit the max limit (something like ~128kB).

We're working on reducing the size.

**Comment by stbenjam** [comment]
/test pj-rehearse

It might rehearse now.

**Comment by jluhrsen** [comment]
/test pj-rehearse

it did rehearse now. Thanks @stbenjam. how did you get it working?

**Comment by stbenjam** [comment]
https://github.com/openshift/release/pull/17445 and https://github.com/openshift/release/pull/18583 made our jobs a lot smaller, we removed ~150 test skips. We've been slowly working our way through the conformance tests to make them work on IPv6/disconnected/baremetal, that was the latest batch.

The one test failure you see will be fixed by https://github.com/openshift-metal3/dev-scripts/pull/1242 and the MCO PR it links to.

**Comment by jluhrsen** [comment]
> #17445 and #18583 made our jobs a lot smaller, we removed ~150 test skips. We've been slowly working our way through the conformance tests to make them work on IPv6/disconnected/baremetal, that was the latest batch.
> 
> The one test failure you see will be fixed by [openshift-metal3/dev-scripts#1242](https://github.com/openshift-metal3/dev-scripts/pull/1242) and the MCO PR it links to.

good stuff. thanks @stbenjam

@vrutkovs , can we get this approved please?

**Comment by vrutkovs** [comment]
`[sig-arch][Early] Managed cluster should start all core operators [Skipped:Disconnected] [Suite:openshift/conformance/parallel]` fails:
>fail [github.com/onsi/ginkgo@v4.7.0-origin.0+incompatible/internal/leafnodes/runner.go:113]: May 22 18:54:35.651: Some cluster operators are not ready: kube-apiserver (Upgradeable=False FeatureGates_RestrictedFeatureGates_IPv6DualStackNoUpgrade: FeatureGatesUpgradeable: "IPv6DualStackNoUpgrade" does not allow updates)

Sounds like we need a test skip / option for openshift-tests to test failure. @jluhrsen please make sure you file a bug for that.

Other than that it looks good,
/approve

**Comment by stbenjam** [comment]
Failure is just because https://github.com/openshift/machine-config-operator/pull/2580 hasn't landed yet, this needs a rebase but otherwise should be good to go

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml
Comment (stbenjam) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:19306]: Were these intentional changes?

Comment (jluhrsen) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml]: yes, this was an accident somehow. New force push just now to get it right and rebase on latest master

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Comment by jluhrsen** [comment]
@vrutkovs , @stbenjam, this seems fine now. the only tests [failing](https://bugzilla.redhat.com/show_bug.cgi?id=1973995) are soon to be [skipped](https://github.com/openshift/origin/pull/26251).

do we need to /override the failing job?

**Comment by vrutkovs** [comment]
LGTM, lets add the same test for 4.9 nightlies too?

**Comment by jluhrsen** [comment]
> LGTM, lets add the same test for 4.9 nightlies too?

yep. just updated to do this.

**Comment by jluhrsen** [comment]
> /approve

I think we need another /lgtm since we lost it when I added the same job for 4.9 on this PR

**Pull request merged**
